fhir-persistence-schema/docs/DB2SchemaMigration.md

```diff
@@ -0,0 +1,382 @@
+
+## Managing the SEQUENCES
+
+Each tenant has a two tenant specific sequences. These sequences are created a single time, and are NOT replaced or upgraded at any point in time. 
+
+| Sequence Name | Description |
+|----------|---------------------------------------------|
+| `fhir_sequence` | The logical id in the database, used for storing each FHIR Resource |
+| `fhir_ref_sequence` | The reference sequence is a logical id for Parameter Names, Code Systems and Resource Types tables |
+
+The stored procedures `add_resource_type`, `add_parameter_name`, `add_code_system`, and `add_any_resource` use the value to uniquely store the resource or supporting data element. 
+
+Do not remove any of the tenant sequences, the sequence must provide unique, non-cycled integer values for the persistence layer to work properly. 
+
+## Managing TABLE 
+
+Each resource has a RESOURCE and LOGICAL RESOURCE table. `_LOGICAL_RESOURCES` has one record per logical resource.  `_RESOURCES` has one record for each version of each logical resource. 
+
+For example, VISIONPRESCRIPTION has: 
+
+```
+VISIONPRESCRIPTION_RESOURCES
+VISIONPRESCRIPTION_LOGICAL_RESOURCES
+```
+
+The data definition has several reference tables to support FHIR Search for each Resoruce type.
+
+| Search Parameter Types | Search Value Table | Description |
+|----------|----------|---------------------------------------------|
+| [Number](https://www.hl7.org/fhir/r4/search.html#number) | **`<RESOURCE>`_NUMBER_VALUES** | Numerical Search for FHIR |
+| [Date/DateTime](https://www.hl7.org/fhir/r4/search.html#date) | **`<RESOURCE>`_DATE_VALUES** | Date, DateTime Search for FHIR |
+| [String](https://www.hl7.org/fhir/r4/search.html#string)<br> [Uri](https://www.hl7.org/fhir/r4/search.html#uri)<br> [Reference](https://www.hl7.org/fhir/r4/search.html#reference)| **`<RESOURCE>`_STR_VALUES** | String Search for FHIR |
+| [Token](https://www.hl7.org/fhir/r4/search.html#token) | **`<RESOURCE>`_TOKEN_VALUES** | Token Search for FHIR |
+| [Composite](https://www.hl7.org/fhir/r4/search.html#composite) | **`<RESOURCE>`_COMPOSITES** | Token Search for FHIR |
+| [Quantity](https://www.hl7.org/fhir/r4/search.html#quantity) | **`<RESOURCE>`_QUANTITY_VALUES** | Quantity Search for FHIR |
+| [Positional](https://www.hl7.org/fhir/r4/location.html#positional) | **`<RESOURCE>`_LATLNG_VALUES** | Positional (near) Search for FHIR |
+
+For instance, for VISIONPRESCRIPTION there are: 
+
+```
+VISIONPRESCRIPTION_NUMBER_VALUES
+VISIONPRESCRIPTION_DATE_VALUES
+VISIONPRESCRIPTION_STR_VALUES
+VISIONPRESCRIPTION_TOKEN_VALUES
+VISIONPRESCRIPTION_COMPOSITES
+VISIONPRESCRIPTION_QUANTITY_VALUES
+VISIONPRESCRIPTION_LATLNG_VALUES
+```
+
+There are also additional tables to support search: `LOGICAL_RESOURCE`, `RESOURCE_TYPES` and `PARAMETER_NAMES` These tables are paritioned, and are specific for each tenant. 
+
+Each of these tables follows a common pattern, and are defined in the `FHIRResourceTableGroup.java`. 
```

Comment:
I suggest removing line 194, or at least moving it elsewhere.  Line 196 seems like it refers back to it, but I think its actually referring to the paragraph before it.

Reply:
cleaned this up.